README.md

```diff
@@ -27,103 +32,103 @@ Then, build the project with:
-```
+This creates an ad-hoc directory structure and populates it with `.cmt`, `.cmi`, `.cmti`, `.mld` from the current switch.
+The `ocamlobjinfo` of the `.cma` files are also copied, as well as the documentation files such as `.md`, `.html` and others (such as the `opam` files). 
 
-Read further for the detailed version of the above commands.
 
-```bash
-$ cd _generated
-$ opam exec -- dune exec -- voodoo-prep
-Warning: No universes have been specified: will generate dummy universes
-$ opam exec -- dune exec -- voodoo-do -p ocaml-base-compiler -b
-$ opam exec -- dune exec -- voodoo-do -p result -b
-$ opam exec -- dune exec -- voodoo-do ...
+## :sweat_drops: 2. Compiling and linking the packages: voodoo-do
+
+Once `voodoo-prep` has run, `voodoo-do` is able to process packages. Simply run:
+
+```sh
+voodoo-do -p <PACKAGE> [-b] [--failed]
 ```
 
-When new packages are installed in the current switch `voodoo-prep` needs to be executed in
-for `voodoo-do` to be able to process them. If the packages are done out of dependency order,
- `voodoo-prep` will alert that there are missing dependencies:
+This command runs `odoc compile` and `odoc link` on the package specified after `-p`.
+If the `--failed` flag is set, a file named `failed` containing `"failed"` is also generated.
 
-```bash
-$ opam exec -- dune exec -- voodoo-do -p odoc -b
+If the packages are processed out of dependency order, `voodoo-do` will alert that there are missing dependencies.
+
+```sh
+voodoo-do -p odoc -b
 ...
 Missing dependency: Stdlib c21c5d26416461b543321872a551ea0d
 ...
 ```
 
 In this case, we need to run `voodoo-do -p ocaml-base-compiler -b` first.
 
-Note that when being used in this mode, the `-b` (blessed) switch should
-always be passed to `voodoo-do`.
+Note that when being used in this mode, the `-b` flag should always be passed to `voodoo-do`.
 
 At this point, to view the output, use `odoc` to generate the support files (mostly: copying `highlight.js` from https://highlightjs.org and `odoc.css`):
 
 ```bash
-$ cd ..
-$ odoc support-files -o _generated/html
+cd ..
+odoc support-files -o html
 ```
 
-and load the package index in your browser (adjust `x.y.z` according to the current switch):
+You can load the package index in your browser:
 
-```bash
-$ open _generated/html/p/ocaml-base-compiler/4.14.0/doc/index.html
+```sh
+open ./html/p/<PACKAGE>/<VERSION>/doc/index.html
 ```
 
-~~An alternative to executing `voodoo-do -p ...` in order is simply to run
-`voodoo-do` with no arguments. This naively executes `voodoo-do` in order
-based on the current switch's dependencies, and is slow and inefficient.~~
+## :deciduous_tree: 3. Rendering the packages: voodoo-gen
 
-To generate the website with the documentation, run:
+Once `voodoo-do` has run, you can generate the website with the documentation, run:
 
 ```bash
-$ opam exec -- dune exec -- voodoo-gen -o output
+voodoo-gen -o <DIR> [-n <PACKAGE>] [--pkg-version <VERSION>]
```

Comment:
What does `--pkg-version` do exactly? I thought every package is only installed once in a switch, and so we only have one version per package after prep?

Note: this is not me implying this is wrong - just me seeing something that didn't match my previous intuition.

Reply:
I thought every package is only installed once in a switch, and so we only have one version per package after prep?

Yes that's also my understanding

> What does `--pkg-version` do exactly?
 
`--pkg-version` provided by ocaml-docs-ci is the version found in the opam file so we check that this version is equal to the one of the prepped package.

Reply:
ok, so it's purely a sanity check?

Reply:
Apparently yes.